File: markdowner-filter/src/test/java/org/jesperancinha/parser/markdowner/filter/SBTFilterTest.java
package org.jesperancinha.parser.markdowner.filter;

import org.junit.jupiter.api.Test;

import java.nio.file.Path;

import static org.assertj.core.api.Assertions.assertThat;

class SBTFilterTest {

    @Test
    public void testSBTFilter() {
        final Path resource = Path.of(getClass().getResource("/directory2NoReadme/project5Sbt/build.sbt").getPath());
        final ProjectFilter<Path> npmFilter = new SBTFilter();

        boolean result = npmFilter.test(resource);

        assertThat(npmFilter.lastProjectName()).isEqualTo("sbt-project");
        assertThat(result).isTrue();
    }

    @Test
    public void testSBTFilterFail() {
        final Path resource = Path.of(getClass().getResource("/directory2NoReadme/project3MavenAndNPM/package.json").getPath());
        final ProjectFilter<Path> mavenFilter = new SBTFilter();

        boolean result = mavenFilter.test(resource);

        assertThat(result).isFalse();
    }
}

File: markdowner-filter/src/test/kotlin/org/jesperancinha/parser/markdowner/filter/SBTFilterTest.kt
package org.jesperancinha.parser.markdowner.filter

import io.kotest.matchers.booleans.shouldBeFalse
import io.kotest.matchers.booleans.shouldBeTrue
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import java.nio.file.Path

internal class SBTFilterTest {
    @Test
    fun testSBTFilter() {
        val buildSbt = javaClass.getResource("/directory2NoReadme/project5Sbt/build.sbt")
        buildSbt.shouldNotBeNull()
        val resource = Path.of(buildSbt.path)
        val npmFilter: ProjectFilter<Path> = SBTFilter()
        val result = npmFilter.test(resource)
        result.shouldBeTrue()
        npmFilter.shouldNotBeNull()
        npmFilter.lastProjectName() shouldBe "sbt-project"
    }

    @Test
    fun testSBTFilterFail() {
        val packageJson = javaClass.getResource("/directory2NoReadme/project3MavenAndNPM/package.json")
        packageJson.shouldNotBeNull()
        val resource = Path.of(packageJson.path)
        val mavenFilter: ProjectFilter<Path> = SBTFilter()
        val result = mavenFilter.test(resource)
        result.shouldBeFalse()
    }
}